_publications/jors/paper.tex

```diff
@@ -0,0 +1,362 @@
+    then launches a temporary version of the proposed website using
+    Surge~\cite{surge}. The PFHub admins can then examine the new
+    submission and further changes can be made if necessary.
+  \item Once review has been completed to the satisfaction of both the
+    uploading scientist and the website maintainers, the pull request
+    is merged and served to the World Wide Web using a hosting service
+    compatible with GitHub Pages.
+\end{itemize}
+
+A combination of Jekyll templates and Coffeescript are used to access
+and download the data links in the submitted YAML files and then
+display the data in interactive plots on the website. The interactive
+plots are displayed using the Plotly JavaScript Graphing
+Library~\cite{plotly} as it provides a programmable interface and
+requires minimal configuration.
+
+The benchmark specifications consist of equations, narrative, plots
+and code samples, and are composed in Jupyter Notebooks. The Jupyter
+Notebooks are included as static objects in the website after
+translation into HTML using the nbconvert tool~\cite{jupyter}.
+
+The combination of a central repository on GitHub for website source
+code and metadata with distributed data records on third-party
+archives avoids the complexity and administrative overhead of
+maintaining a live database and associated back-end application. The
+PFHub infrastructure provides a template for other small scientific
+communities to host custom content and integrate data from members of
+their community.
+
+\begin{figure}
+  \includegraphics[width=\textwidth]{pfhub_website.png}
+  \caption{Schematic overview of the PFHub framework for building
+    scientific research portals, simply.}
+  \centering
+  \label{fig:pfhub_website}
+\end{figure}
+
+\section*{Quality control}
+
+The framework has a fully automated test recipe deployed on Travis CI
+with an environment built using the Nix Package
+Manager~\cite{nix}. The environment is pinned to a specific version of
+the Nixpkgs repository ensuring fully reproducible build and test
+phases as well as ensuring that the development and automated testing
+environments are identical. The full test recipe is outlined in a
+\texttt{.travis.yml} file stored in the repository~\cite{travisyml}
+and consists of the following steps.
+
+\begin{itemize}
+  \item Build the Nix environment from a cached storage reducing the
+    build time.
+  \item Run automated tests on Jupyter Notebooks using
+    NBval~\cite{nbval} and Py.test~\cite{pytest}.
+  \item Run validation tests on HTML files using
+    HTMLProofer~\cite{htmlproofer}.
+  \item Lint and test front-end Coffeescript using appropriate tools.
+  \item Display a temporary version of the website using
+    Surge~\cite{surge} for visual review.
+\end{itemize}
+
+\section*{(2) Availability}
+\vspace{0.5cm}
+\section*{Operating system}
+
+The PFHub framework can be deployed on any platform supporting Nix,
```

Comment:
Again, topic sentence.

Also, what would someone who is running Windows do?  Suggest a solution.

Reply:
I mentioned that the deployment is platform independent since we don't have servers so not an issue for deploying. Developers would need to use the Docker container on Windows, which is now in the text.